Doc/library/ipaddress.rst

```diff
@@ -199,9 +199,34 @@ write code that handles both IP versions correctly.  Address objects are
       ``True`` if the address is reserved for link-local usage.  See
       :RFC:`3927`.
 
+   .. method:: __format__(fmt)
+
+      Returns a string representation of the IP address, controlled by
+      an explicit format string.
+      *fmt* can be one of the following: ``'s'``, the default option,
+      equivalent to :func:`str`, ``'b'`` for a zero-padded binary string,
+      ``'X'`` or ``'x'`` for an uppercase or lowercase hexadecimal 
+      representation, or ``'n'``, which is equivalent to ``'b'`` for IPv4
+      addresses and ``'x'`` for IPv6. For binary and hexadecimal
+      representations, the form specifier ``'#'`` and the grouping option
+      ``'_'`` are available.
+
+         >>> ipaddress.IPv4Address('192.168.0.1').__format__('s')
```

Comment:
Having said that I think the documentation belongs under `__format__` (which I still think is the right thing to do), I'm not sure using `__format__` in the examples is the right thing. We discourage people from calling dunders directly. Maybe there should be a note that `__format__` is used by `format`, `str.format`, and f-strings, and then show examples using one or more of those? I'm open to suggestions here.